content/axon-server-installation-guides/modules/ROOT/pages/professional/kubernetes.adoc

```diff
@@ -1,5 +1,350 @@
+pass:[<!-- vale AxonIQ.Headings = NO -->]
+= Kubernetes
 
-:page-needs-improvement: content
-:page-needs-content: We are working on completing this guide for installing Axon Server on Kubernetes. Please check back later.
+Kubernetes has become an industry standard when it comes to deploying Software.
+While we recommend installing Axon Server on virtual machines due to the better stability they provide, this guide will help you in installing Axon Server Cluster on Kubernetes.
 
-= Kubernetes
\ No newline at end of file
+[NOTE]
+====
+AxonIQ does not provide a HELM chart, nor any installation instructions for service meshes such as `istio`.
+You can adapt the instruction in this guide for your specific situation.
+====
+
+== Overview
+
+The Kubernetes setup works with a variety of individual parts:
+
+. *Stateful set*: The stateful set for each node runs Axon Server with a stable hostname, necessary for the headless services.
+. *Persistent Volume Claims*: As Axon Server needs to store data, there are several PVC's necessary.
+These are included in the StatefulSet configuration.
+. *Network Service*: For applications to be able to reach Axon Server and communication between nodes, Kubernetes requires a network service that resolves to the pod created by the stateful set.
+. *Config Map*: A config map with the properties for Axon Server's configuration.
+
+== Step 1: Create a namespace
+
+We recommend running Axon Server in its separate namespace.
+During this guide, we will use the `axonserver` namespace.
+Because of Kubernetes networking, the namespace name selected affects some of the properties in your property file, such as `axoniq.axonserver.domain` and `axoniq.axonserver.internal-domain`.
+
+Create it using the following command:
+
+[source,bash]
+----
+kubectl create namespace axonserver
+----
+
+[NOTE]
+====
+You can name this namespace any way you like.
+However, if you modify this, make sure you modify commands and properties later in this tutorial as well.
+====
+
+== Step 1: Choose your image (optional)
+
+include::partial$choose-your-docker-image.adoc[]
+
+== Step 2: AxonIQ Console
+In order to run a cluster of Professional nodes, you need to connect it to AxonIQ Console, after enabling your subscription.
+
+By connecting your Axon Server Professional to AxonIQ Console you will unlock additional functionality.
+
+.Why connect to AxonIQ Console?
+[%collapsible]
+====
+include::partial$console-additional-functionality.adoc[]
+====
+
+.Obtaining AxonIQ Console access token
+[%collapsible]
+====
+include::partial$obtain-console-access-token.adoc[]
+====
+
+== Step 3: Configure properties
+
+When connecting to AxonIQ Console, your server will receive all information automatically.
```

Comment:
```suggestion
When connecting to AxonIQ Console, your server will automatically receive information to form the cluster. This does require the hostname information to be set correctly; otherwise the nodes won't be able to find each other.
```